serializable.ts

```diff
@@ -179,42 +179,33 @@ function fromJSON<T>(
   const serializablePropertyMap = SERIALIZABLE_CLASS_MAP.get(
     context?.constructor?.prototype,
   );
+
   if (!serializablePropertyMap) {
     throw new Error(
       `${ERROR_MISSING_PROPERTIES_MAP}: ${context?.constructor
         ?.prototype}`,
     );
   }
 
-  const _json = typeof json === "string" ? json : JSON.stringify(json);
+  const _json = typeof json === "string" ? JSON.parse(json) : json;
+  const accumulator: Partial<T> = {};
+
+  for (const [key, value] of Object.entries(_json)) {
+    const {
+      propertyKey,
+      fromJSONStrategy = fromJSONDefault,
+    } = serializablePropertyMap.getBySerializedKey(key) || {};
+    const processedValue: unknown = Array.isArray(value)
+      ? value.map((v) => fromJSONStrategy(v))
+      : fromJSONStrategy(value as JSONValue);
```

Comment:
Should we nest this inside the `if (propertyKey)` below, so we don't process it if it's not a `@SerializeProperty()`?